### PR TITLE
Update documentation with example of base_path proc

### PR DESCRIPTION
**Pull request by Lordnibbler**
per https://github.com/ruby-grape/grape-swagger/pull/441#issuecomment-222338147

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,7 +202,12 @@ You can pass a hash with optional configuration settings to ```add_swagger_docum
 *not all configuration options supported yet*, but is WIP
 
 
-`host` and `base_path` are also accepting a `proc` to evaluate.
+The `host` and `base_path` options also accept a `proc` or `lambda` to evaluate, which is passed a request object:
+
+```ruby
+add_swagger_documentation \
+  base_path: proc { |foo| foo.host =~ /^example/ ? '/api-example' : '/api' }
```
Comment (dblock) [inline README.md]: `foo` here should be `request`

File: README.md [hunk 2]
```diff
@@ -202,7 +202,12 @@ You can pass a hash with optional configuration settings to ```add_swagger_docum
 *not all configuration options supported yet*, but is WIP
 
 
-`host` and `base_path` are also accepting a `proc` to evaluate.
+The `host` and `base_path` options also accept a `proc` or `lambda` to evaluate, which is passed a request object:
```
Comment (dblock) [inline README.md]: change request to the class name linked to the Rack doc of the type of request

**Comment by Lordnibbler** [comment]
@dblock updated! Thanks for the suggestions. I also updated the test where I copied that `foo` request object from!

**Comment by dblock** [comment]
Thanks, merged.

**Pull request merged**